### PR TITLE
Richard woessner code review1

### PR DESCRIPTION
**Pull request by Richard-Woessner**
## Analysis of the program
RedCord right now stores messages and will display all collected message data in their /allMessages endpoint. It was easy to work with and structured well, but I do not know where the application is going.

#### Was the program available in UC Github on time?
Yes the program was available to me on the 4th.

#### Is the program easy for you, as an outsider, to understand?

I had a easy experience with working on the application itself. Everything was layered well and all the classes were named properly. As for what the app will be doing in the future I had a hard time understanding. 

#### Does the program compile?
Yes

#### Three technical concepts
In RedCord Service stub they had both the Service and Component annotations, I learned that Service is a specialized version of Component, so there is no need to have both. I removed the Component annotation.

```
@Service
@Component
```

If you want to use a key value array in Java HashMap is what would be used. I found it cool that you are not limited to the String or integers for the Keys, you could any object as the key.

`Map<Integer,` Topic> allTopics = new `HashMap<>();`

I originally wanted to hash the password for the User class, and I assumed that Spring Boot would have a built-in password handling annotation. I assumed this because if SpringBoot creates setters and getters which are almost universal for data handling. Salting and hashing passwords are very basic and uniform for password handling, so I assumed that SpringBoot would implement those that just like the Getters and Setters.

```
@SerializedName("userId") private Integer userId;
@SerializedName("userName") private String userName;
@SerializedName("password") private String password;
@SerializedName("email") private String email;
@SerializedName("status") private String status;
```

### CODE REVIEW

File: src/main/java/com/enterprise/redcord/RedCordController.java [hunk 1]
```diff
@@ -21,17 +21,21 @@
...
-    
-    
-    
+
+
+    /**
+     * @param topic from data containing user created Topic
+     * @return /start
+     */
+    //Endpoint called
```
Comment (Richard-Woessner) [inline src/main/java/com/enterprise/redcord/RedCordController.java:38]: Added JavaDoc to better understand endpoint functionality

File: src/main/java/com/enterprise/redcord/RedCordController.java [hunk 2]
```diff
@@ -21,17 +21,21 @@
 
     /**
      * Handle the root (/) endpoint and return a start page
-     * @return
+     * @return /start
```
Comment (Richard-Woessner) [inline src/main/java/com/enterprise/redcord/RedCordController.java:24]: Added JavaDoc to better understand endpoint functionality

File: src/main/java/com/enterprise/redcord/RedCordController.java [hunk 3]
```diff
@@ -42,6 +46,10 @@ public String saveTopic(Topic topic) {
         return "start";
     }
 
+    /**
+     * @param messageEntry Message created via form on index page
+     * @return /start
+     */
```
Comment (Richard-Woessner) [inline src/main/java/com/enterprise/redcord/RedCordController.java:52]: Added JavaDoc to better understand endpoint functionality

File: src/main/java/com/enterprise/redcord/RedCordController.java [hunk 4]
```diff
@@ -52,6 +60,11 @@ public String saveMessage(Message messageEntry) {
         return "start";
     }
 
+
+    /**
+     * The /allMessages enpoint returns a JSON file containing all Message data stored.
+     * @return JSON page containing all Message data via fetchAllMessages()
+     */
```
Comment (Richard-Woessner) [inline src/main/java/com/enterprise/redcord/RedCordController.java:67]: Added JavaDoc to better understand endpoint functionality

File: src/main/java/com/enterprise/redcord/dao/IUserDAO.java [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+import com.enterprise.redcord.dto.User;
+
+import java.util.List;
+
+public interface IUserDAO {
+
+    User saveUser(User user) throws Exception;
+
+    List<User> fetchAllUsers();
+}
```
Comment (Richard-Woessner) [inline src/main/java/com/enterprise/redcord/dao/IUserDAO.java:12]: Added IUserDAO with ReadCord naming convention

File: src/main/java/com/enterprise/redcord/dao/UserDAOStub.java [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    @Override
+    public User saveUser(User userEntry) throws Exception {
+        return null;
+    }
+
+    @Override
+    public List<User> fetchAllUsers() {
+        return null;
+    }
+}
```
Comment (Richard-Woessner) [inline src/main/java/com/enterprise/redcord/dao/UserDAOStub.java:24]: Added IUserDAO with ReadCord naming convention

File: src/main/java/com/enterprise/redcord/service/IUserService.java [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+import com.enterprise.redcord.dto.User;
+import java.util.List;
+
+
+public interface IUserService {
+
+    User saveUser(User user) throws Exception;
+
+    List<User> fetchAllUsers();
+}
```
Comment (Richard-Woessner) [inline src/main/java/com/enterprise/redcord/service/IUserService.java:12]: Added User service layer with ReadCord naming convention

File: src/main/java/com/enterprise/redcord/service/UserServiceStub.java [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+    @Override
+    public User saveUser(User user) throws Exception {
+        return null;
+    }
+
+    @Override
+    public List<User> fetchAllUsers() {
+        return null;
+    }
+}
```
Comment (Richard-Woessner) [inline src/main/java/com/enterprise/redcord/service/UserServiceStub.java:19]: Added User service layer with ReadCord naming convention

File: src/main/java/com/enterprise/redcord/service/TopicServiceStub.java [hunk 9]
```diff
@@ -8,7 +8,6 @@
 
 
 @Service
-@Component
```
Comment (Richard-Woessner) [inline src/main/java/com/enterprise/redcord/service/TopicServiceStub.java:11]: Removed redundant Component

File: src/main/java/com/enterprise/redcord/dao/TopicDAOStub.java [hunk 10]
```diff
@@ -8,7 +8,6 @@
 import java.util.Map;
 
 @Repository
-@Component
```
Comment (Richard-Woessner) [inline src/main/java/com/enterprise/redcord/dao/TopicDAOStub.java:11]: Removed redundant Component

File: src/main/java/com/enterprise/redcord/RedCordController.java (hunk 1 above)
Comment (discospiff) [reply src/main/java/com/enterprise/redcord/RedCordController.java:38]: Remove // after the JavaDoc.

File: src/main/java/com/enterprise/redcord/service/UserServiceStub.java [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+public class UserServiceStub implements IUserService{
+    @Override
+    public User saveUser(User user) throws Exception {
+        return null;
+    }
+
+    @Override
+    public List<User> fetchAllUsers() {
+        return null;
+    }
```
Comment (discospiff) [inline src/main/java/com/enterprise/redcord/service/UserServiceStub.java:18]: This appears to be an incomplete feature.  In a code review, focus on reducing technical debt of existing code.  Avoid adding new, incomplete functionality .
1) Only features that meet the Definition of Done should be merged to Master.
2) Unused code is clutter, which increases technical debt.

**Comment by discospiff** [comment]
From this review, I'd harvest the JavaDoc.  Either manually, or through merge request.

**Comment by dmaddx** [comment]
Will implement JavaDocs on working features. Features not yet complete.

**Pull request closed without merging**